kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkMicroBatchStream.java

```diff
@@ -614,6 +703,127 @@ private void validateCommit(
+
+      // There is a schema change. All files after this commit will use `schemaChange`. Hence, we
+      // check whether we can use `schema` (the fixed source schema we use in the same run of the
+      // query) to read these new files safely.
+      boolean backfilling = version < snapshotAtSourceInit.getVersion();
+      // We forbid the case when the schemaChange is nullable while the read schema is NOT
+      // nullable, or in other words, `schema` should not tighten nullability from `schemaChange`,
+      // because we don't ever want to read back any nulls when the read schema is non-nullable.
+      boolean shouldForbidTightenNullability = !forceEnableUnsafeReadOnNullabilityChange;
+      // If schema tracking is disabled for type widening, we allow widening type changes to go
+      // through without requiring the user to set `allowSourceColumnTypeChange`. The schema change
+      // will cause the stream to fail with a retryable exception, and the stream will restart using
+      // the new schema.
+      boolean allowWideningTypeChanges = typeWideningEnabled; // TODO(#5319): schema tracking
+      // If a user is streaming from a column mapping table and enable the unsafe flag to ignore
+      // column mapping schema changes, we can allow the standard check to allow missing columns
+      // from the read schema in the schema change, because the only case that happens is when
+      // user rename/drops column but they don't care so they enabled the flag to unblock.
+      // This is only allowed when we are "backfilling", i.e. the stream progress is older than
+      // the analyzed table version. Any schema change past the analysis should still throw
+      // exception, because additive schema changes MUST be taken into account.
+      boolean shouldAllowMissingColumns =
+          isStreamingFromColumnMappingTable
+              && allowUnsafeStreamingReadOnColumnMappingSchemaChanges
+              && backfilling;
+      // Partition column change will be ignored if user enable the unsafe flag
+      Seq<String> newPartitionColumns, oldPartitionColumns;
+      if (allowUnsafeStreamingReadOnPartitionColumnChanges) {
+        newPartitionColumns = (Seq<String>) Seq$.MODULE$.empty();
+        oldPartitionColumns = (Seq<String>) Seq$.MODULE$.empty();
+      } else {
+        newPartitionColumns =
+            CollectionConverters.asScala(
+                    VectorUtils.toJavaList(newMetadata.getPartitionColumns()).stream()
+                        .map(Object::toString)
+                        .collect(Collectors.toList()))
+                .toSeq();
+        oldPartitionColumns =
+            CollectionConverters.asScala(
+                    VectorUtils.toJavaList(oldMetadata.getPartitionColumns()).stream()
+                        .map(Object::toString)
+                        .collect(Collectors.toList()))
+                .toSeq();
+      }
+
+      final var result =
```

Comment:
Can we just declare its actual classes and not using var? It is hard to infer what the actual type